packages/components/src/styles/components/form/key-value-inputs.scss

```
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// FORM > KEY-VALUE-INPUTS
+//
+
+@use "../../mixins/breakpoints" as *;
+@use "../../mixins/screen-reader-only" as *;
+
+// notice: this is a <fieldset> element
+.hds-form-key-value-inputs {
+  display: flex;
+  flex-direction: column;
+  margin: 0;
+  padding: 0;
+  border: none;
+  row-gap: 24px;
+
+  @include hds-breakpoint-above("md") {
+    row-gap: 12px;
+  }
+
+  &:focus {
+    // we move focus to the fieldset when a row is deleted, but we don't need the focus outline visible.
+    outline: none;
+  }
+}
+
+.hds-form-key-value-inputs__row {
+  display: flex;
+  flex-direction: column;
+  row-gap: 8px;
+  align-items: flex-start;
+
+  @include hds-breakpoint-above("md") {
+    display: grid;
+    flex-grow: 1;
+    grid-template-columns: var(--hds-key-value-pair-columns, 1fr 1fr min-content);
+    align-items: flex-end;
+    column-gap: 8px;
+    row-gap: 0;
+
+    &.hds-form-key-value-inputs__row--first {
+      .hds-form-key-value-inputs__delete-row-button-container,
+      .hds-form-key-value-inputs__generic-container {
+        grid-row: 2;
+        // need to set the `grid-column` because the `grid-row` is set, otherwise it defaults to the first column
+        grid-column: var(--hds-key-value-inputs-column-index);
+      }
+    }
+
+    &:not(.hds-form-key-value-inputs__row--first) {
+      .hds-form-key-value-inputs__field-header {
+        @include screen-reader-only();
+      }
+    }
+  }
+}
+
+.hds-form-key-value-inputs__footer {
+  display: flex;
+  flex-direction: column;
+  row-gap: 12px;
+  align-items: flex-start;
+}
+
+.hds-form-key-value-inputs__field {
+  display: grid;
+  grid-row: span 2;
+  grid-template-rows: subgrid;
+  width: 100%; // we want the input element to fill the parent container
+
+  .hds-form-key-value-inputs__field-header {
+    align-self: flex-end;
+    min-width: 0;
+    margin-bottom: 12px; // space between label and control
```

```suggestion
    margin-bottom: 8px; // space between label and control
```
[Suggestion] In @majedelass 's previous comment I think they said the gap between the label and input should be 8px on mobile.